svd-encoder/src/config.rs

```diff
@@ -0,0 +1,472 @@
+use std::str::FromStr;
+
+use convert_case::{Boundary, Case, Casing};
+
+use crate::svd::BitRangeType;
+
+#[derive(Clone, Copy, Debug)]
+pub enum IdentifierFormat {
+    /// `Camel` case
+    ///
+    /// `aCamelCaseExample`
+    Camel,
+    /// `Pascal` case
+    ///
+    /// `APascalCaseExample`
+    Pascal,
+    /// `Snake` case
+    ///
+    /// `a_snake_case_example`
+    Snake,
+    /// `UpperSnake` case
+    ///
+    /// `AN_UPPER_SNAKE_CASE_EXAMPLE`
+    UpperSnake,
```

Comment:
Here http://calebmer.com/inflections/inflections/case/index.html
it is called "constant case"